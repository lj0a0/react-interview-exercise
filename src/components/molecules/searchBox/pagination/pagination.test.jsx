import React from "react";
import Pagination from "./pagination";
import renderer from "react-test-renderer";


test("Pagination snapshot first page", () => {
  const component = renderer.create(
    <Pagination
      pagesSearch={{
        current: "1",
        first: "",
        next: "2",
        prev: "",
        last: "34"
      }}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Pagination snapshot last page", () => {
  const component = renderer.create(
    <Pagination
      pagesSearch={{
        current: "34",
        first: "1",
        next: "",
        prev: "33",
        last: ""
      }}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Pagination snapshot Middle", () => {
  const component = renderer.create(
    <Pagination
      pagesSearch={{
        current: "5",
        first: "1",
        next: "6",
        prev: "4",
        last: "34"
      }}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Pagination snapshot Null", () => {
  const component = renderer.create(
    <Pagination
      pagesSearch={{
        current: "",
        first: "",
        next: "",
        prev: "",
        last: ""
      }}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

