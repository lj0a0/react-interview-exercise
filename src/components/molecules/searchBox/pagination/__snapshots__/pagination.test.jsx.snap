// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination snapshot Middle 1`] = `
<div
  className="col-md-6"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item"
      style={
        Object {
          "marginRight": 10,
        }
      }
    >
      <button
        className="page-link"
        onClick={[Function]}
      >
        First
      </button>
    </li>
    <li
      className="page-item"
    >
      <button
        className="page-link"
      >
        Prev
      </button>
    </li>
    <li
      className="page-item"
    >
      <button
        className="page-link"
        onClick={[Function]}
      >
        4
      </button>
    </li>
    <li
      className="page-item"
    >
      <button
        className="page-link"
        onClick={[Function]}
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        5
      </button>
    </li>
    <li
      className="page-item"
    >
      <button
        className="page-link"
        onClick={[Function]}
      >
        6
      </button>
    </li>
    <li
      className="page-item"
    >
      <button
        className="page-link"
        onClick={[Function]}
      >
        Next
      </button>
    </li>
    <li
      className="page-item"
      style={
        Object {
          "marginLeft": 10,
        }
      }
    >
      <button
        className="page-link"
        onClick={[Function]}
      >
        Last
      </button>
    </li>
  </ul>
</div>
`;

exports[`Pagination snapshot Null 1`] = `null`;

exports[`Pagination snapshot first page 1`] = `
<div
  className="col-md-6"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item"
    >
      <button
        className="page-link"
        onClick={[Function]}
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        1
      </button>
    </li>
    <li
      className="page-item"
    >
      <button
        className="page-link"
        onClick={[Function]}
      >
        2
      </button>
    </li>
    <li
      className="page-item"
    >
      <button
        className="page-link"
        onClick={[Function]}
      >
        3
      </button>
    </li>
    <li
      className="page-item"
    >
      <button
        className="page-link"
        onClick={[Function]}
      >
        Next
      </button>
    </li>
    <li
      className="page-item"
      style={
        Object {
          "marginLeft": 10,
        }
      }
    >
      <button
        className="page-link"
        onClick={[Function]}
      >
        Last
      </button>
    </li>
  </ul>
</div>
`;

exports[`Pagination snapshot last page 1`] = `
<div
  className="col-md-6"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item"
      style={
        Object {
          "marginRight": 10,
        }
      }
    >
      <button
        className="page-link"
        onClick={[Function]}
      >
        First
      </button>
    </li>
    <li
      className="page-item"
    >
      <button
        className="page-link"
        onClick={[Function]}
      >
        Prev
      </button>
    </li>
    <li
      className="page-item"
    >
      <button
        className="page-link"
        onClick={[Function]}
      >
        33
      </button>
    </li>
    <li
      className="page-item"
    >
      <button
        className="page-link"
        onClick={[Function]}
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        34
      </button>
    </li>
  </ul>
</div>
`;
